ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -919,6 +934,37 @@ public static boolean getBooleanHeader(Map<String, Object> args, String header,
     return booleanValue;
   }
 
+  /**
+   * Gets the value of a header as a {@code boolean}.
+   * @param args a map of arguments to be used to look for {@code header}.
+   * @param header the name of the header.
+   * @param required if {@code true}, {@link RestServiceException} will be thrown if {@code header} is not present
+   *                 in {@code args}.
+   * @param defaultVal the default value for this header
+   * @return {@code true} if the header's value is {@code "true"} (case-insensitive), or {@code false} if the header's
+   *         value is {@code "false} (case-insensitive) or the header is not present and {@code required} is
+   *         {@code false}.
+   * @throws RestServiceException same as cases of {@link #getHeader(Map, String, boolean)} and if the value cannot be
+   *                              converted to a {@code boolean}.
+   */
+  public static boolean getBooleanHeaderWithDefaultValue(Map<String, Object> args, String header, boolean required, boolean defaultVal)
```

Comment:
My point of making a new function was to 1) not cascade your changes into existing code path 2) also reduce the number of arguments to the function itself to make the code readable. 
I don't prefer names like XXwithYY because later on someone will add a new arg and call it XXwithYYandZZ etc. Once you name it `getOptionalBooleanHeader`, you automatically get rid of the redundant `required` arg and then passing a `default` value follows from the name. I hope going forward people will use it. The original method should have been called `getRequiredBooleanHeader` but that's history. But going forward, the names and function signatures should be short, meaningful and descriptive at a glance.

Reply:
Sounds good, updated it in latest commit, can you take another look?